File: lab7/DictionaryTest.java
//-----------------------------------------------------------------------------
// DictionaryTest.java
//-----------------------------------------------------------------------------
// Name:        Ahmed Elshaarany
// CruzID:      aelshaar
// Class:       12M
// Assignment:  lab7
//-----------------------------------------------------------------------------
// Description:
// This file tests the Dictionary ADT
//----------------------------------------------------------------------------

public class DictionaryTest{

    public static void main(String[] args){

      String v;
      Dictionary A = new Dictionary();

      // ------------------ Testing isEmpty() and size() methods ---------------------//
      /*
      System.out.println(A.isEmpty());
      System.out.println(A.size());
      */

      // ------------------ Testing insert(), lookup() and toString() methods ---------------------//
      A.insert("1","a");
      A.insert("2","b");
      A.insert("3","c");
      A.insert("4","d");
      A.insert("5","e");
      A.insert("6","f");
      A.insert("7","g");
      System.out.println(A);

      v = A.lookup("1");
      System.out.println("key=1 "+(v==null?"not found":("value="+v)));
      v = A.lookup("3");
      System.out.println("key=3 "+(v==null?"not found":("value="+v)));
      v = A.lookup("7");
      System.out.println("key=7 "+(v==null?"not found":("value="+v)));
      v = A.lookup("8");
      System.out.println("key=8 "+(v==null?"not found":("value="+v)));
      System.out.println();

      //A.insert("2","f");  // causes KeyCollisionException

      // ------------------ Testing insert(), lookup(), toString(), and delete()  methods ---------------------//

      A.delete("1");
      A.delete("3");
      A.delete("7");
      System.out.println(A);
      
      //A.delete("8");  // causes KeyNotFoundException
      
      
      // ------------------ Testing the rest of the DictionaryClient file ---------------------//

      System.out.println(A.isEmpty());
      System.out.println(A.size());
      A.makeEmpty();
      System.out.println(A.isEmpty());
      System.out.println(A);
      
    }
    

}

File: pa5/DictionaryTest.c
//-----------------------------------------------------------------------------
// DictionaryTest.c
//----------------------------------------------------------------------------                           
// Name:        Ahmed Elshaarany                                                                         
// CruzID:      aelshaar                                                                                 
// Class:       12B                                                                                      
// Assignment:  pa5                                                                                      
//-----------------------------------------------------------------------------                          
// Description:                                                                                          
// Test C file for the Dictionary ADT
//-----------------------------------------------------------------------------                          

#include<stdio.h>
#include<stdlib.h>
#include<assert.h>
#include"Dictionary.h"

int main(int argc, char* argv[]){


  Dictionary A = newDictionary();

  // ----------------------------- Testing insert(), isEmpty(), and size() functions ---------------------------------- //

  /*
  printf("Dictionary init size = %d\n", size(A));
  printf("Is Dictionary emtpy at the begining? %s\n", (isEmpty(A)==1)? "Yes":"No");
  //  char* k;
  //  char* v;
  char* word1[] = {"one","two","three","four","five","six","seven"};
  char* word2[] = {"foo","bar","blah","galumph","happy","sad","blue"};
  int i;
  
  for(i=0; i<7; i++){
    insert(A, word1[i], word2[i]);
  }

  printf("Dictionary size now = %d\n", size(A));
  printf("Is Dictionary emtpy now? %s\n", (isEmpty(A)==1)? "Yes":"No");
  */

  // ----------------------------- Testing insert() and printDictionary() functions ---------------------------------- //

  /*
  char* word1[] = {"one","two","three","four","five","six","seven"};
  char* word2[] = {"foo","bar","blah","galumph","happy","sad","blue"};
  int i;
  
  for(i=0; i<7; i++){
    insert(A, word1[i], word2[i]);
  }
  
  printDictionary(stdout, A);
  */
  

  // ----------------------------- Testing insert(), printDictionary(), lookup functions ---------------------------------- //

  /*
  char* k;
  char* v;
  char* word1[] = {"one","two","three","four","five","six","seven"};
  char* word2[] = {"foo","bar","blah","galumph","happy","sad","blue"};
  int i;
  
  for(i=0; i<7; i++){
    insert(A, word1[i], word2[i]);
  }
  
  printDictionary(stdout, A);
  
  for(i=0; i<7; i++){
    k = word1[i];
    v = lookup(A, k);
    printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
  }

  //insert(A, "five", "glow"); // error: duplicate keys
  */


  // ------------ Testing insert(), printDictionary(), lookup(), delete() functions ---------------------- //

  /*
  char* k;
  char* v;
  char* word1[] = {"one","two","three","four","five","six","seven"};
  char* word2[] = {"foo","bar","blah","galumph","happy","sad","blue"};
  int i;
  
  for(i=0; i<7; i++){
    insert(A, word1[i], word2[i]);
  }
  
  printDictionary(stdout, A);
  
  for(i=0; i<7; i++){
    k = word1[i];
    v = lookup(A, k);
    printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
  }
  
  // insert(A, "five", "glow"); // error: duplicate keys
  
  delete(A, "one");
  delete(A, "three");
  delete(A, "seven");
  
  printDictionary(stdout, A);
  
  for(i=0; i<7; i++){
    k = word1[i];
    v = lookup(A, k);
    printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
  }
  
  // delete(A, "one");  // error: key not found
  
  printf("%s\n", (isEmpty(A)?"true":"false"));
  printf("%d\n", size(A));
  */


  // ------------ Testing all of DictionaryClient.c ---------------------- //
  
  char* k;
  char* v;
  char* word1[] = {"one","two","three","four","five","six","seven"};
  char* word2[] = {"foo","bar","blah","galumph","happy","sad","blue"};
  int i;
  
  for(i=0; i<7; i++){
    insert(A, word1[i], word2[i]);
  }
  
  printDictionary(stdout, A);
  
  for(i=0; i<7; i++){
    k = word1[i];
    v = lookup(A, k);
    printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
  }
  
  // insert(A, "five", "glow"); // error: duplicate keys
  
  delete(A, "one");
  delete(A, "three");
  delete(A, "seven");
  
  printDictionary(stdout, A);
  
  for(i=0; i<7; i++){
    k = word1[i];
    v = lookup(A, k);
    printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
  }
  
  // delete(A, "one");  // error: key not found
  
  printf("%s\n", (isEmpty(A)?"true":"false"));
  printf("%d\n", size(A));
  makeEmpty(A);
  printf("%s\n", (isEmpty(A)?"true":"false"));
  
  freeDictionary(&A);
  
  return (EXIT_SUCCESS);
}
